packages/nx/src/nx-cloud/utilities/url-shorten.ts

```diff
@@ -141,10 +141,38 @@ async function getNxCloudVersion(apiUrl: string): Promise<string | null> {
   } catch (e) {
     logger.verbose(`Failed to get version of Nx Cloud.
       ${e}`);
+    return null;
   }
 }
 
-function removeVersionModifier(versionString: string): string {
+export function removeVersionModifier(versionString: string): string {
   // version may be something like 2406.13.5.hotfix2
   return versionString.split(/[\.-]/).slice(0, 3).join('.');
 }
+
+export function compareCalver(version1: string, version2: string): number {
+  const parseVersion = (version: string) => {
+    const parts = version.split('.').map((part) => parseInt(part, 10));
+    return {
+      // Technically we're not using year, but it follows the same format
+      year: parts[0],
+      month: parts[1],
+      day: parts[2],
+    };
+  };
+
+  const v1 = parseVersion(version1);
+  const v2 = parseVersion(version2);
+
+  if (v1.year !== v2.year) {
+    return v1.year > v2.year ? 1 : -1;
+  }
+  if (v1.month !== v2.month) {
+    return v1.month > v2.month ? 1 : -1;
+  }
+  if (v1.day !== v2.day) {
+    return v1.day > v2.day ? 1 : -1;
+  }
+
+  return 0;
+}
```

Comment:
I think this is technically wrong, we don't use `year.month.day`, we use `YYMM.DD.BUILD`

Reply:
I am pretty sure the comparison results will be correct regardless, but I will update the names of the variables and function to avoid confusion.